app/Http/Middleware/ContentType.php

```diff
@@ -34,15 +34,17 @@ class ContentType
 	 */
 	public function handle(Request $request, \Closure $next, string $contentType): mixed
 	{
-		if ($contentType === self::JSON) {
+		$any = empty($request->getContentType()) || $request->getContentType() === '*/*';
+
+		if ($contentType === self::JSON && !$any) {
 			if (!$request->isJson()) {
 				throw new UnexpectedContentType(self::JSON);
 			}
-		} elseif ($contentType === self::MULTIPART) {
+		} elseif ($contentType === self::MULTIPART && !$any) {
 			if ($request->getContentType() !== 'form') {
 				throw new UnexpectedContentType(self::MULTIPART);
 			}
-		} else {
+		} elseif (!$any) {
```

Comment:
Some problem here. We must be really careful about the content types because there are deeply hidden dependencies to the exception handler. It had cost me hours (if not even days) to read through the Laravel source code and figure that out.

If there is an urgent need to push this PR forward, I would be fine with it, if someone else did the review on the parts which are directly related to the API keys, but please leave the content types untouched in that case.

I also don't really see how this is related to API keys. For me these are two orthogonal things.

If these changes are really necessary (for whatever reason), then please, please wait with this PR until the other PRs are out of the door and I can have a look into this one.